drivers/system_pathways/app/models/system_pathways/chart_base.rb

```diff
@@ -141,13 +144,84 @@ def average(value, count)
       scope.merge(race_scope)
     end
 
+    private def filter_for_race_with_details_filter(scope, race_filter)
+      return scope.where(id: multi_racial_clients(scope).select(:id)) if race_filter.races.include?('MultiRacial')
+
+      query = races.except('MultiRacial', 'RaceNone').keys.map { |k| [k.underscore.to_sym, false] }.to_h
+      race_filter.races.each do |column|
+        next if column.in?(['MultiRacial', 'RaceNone'])
+
+        query[column.underscore.to_sym] = true
+      end
+      scope.where(query)
+    end
+
+    private def filter_for_ethnicity_with_details_filter(scope, filter)
+      unknown_query = races.except('MultiRacial', 'RaceNone').keys.map { |k| [k.underscore.to_sym, false] }.to_h
+
+      # These are mutually exclusive, so, although the filter is an array, ignore multiples
+      case filter.ethnicities.first
+      when :hispanic_latinaeo
+        scope.where(hispanic_latinaeo: true)
+      when :non_hispanic_latinaeo
+        scope.where(hispanic_latinaeo: false).
+          where.not(unknown_query).except(:hispanic_latinaeo) # Must have at least one race response
```

Comment:
This doesn't look correct, should the `except` go on the `unknow_query`?  It looks like active record has an `except` [method](https://apidock.com/rails/v7.0.0/ActiveRecord/SpawnMethods/except) but I don't think it's doing what you intend.
Also, shouldn't unknown be something like `RaceNone in (8, 9, 99)`?